### DISCUSSION TITLE
extended .mpy still runs OK

### DISCUSSION THREAD

**Discussion by kjm1102** [micropython/micropython#15207] [Using MicroPython] [answered]
After a bad experience with a corrupted .mpy OTA download I decided to;
1) manually add a 5 digit file length to the end of the .mpy after creation on the PC
2) check & remove the file length addon after download on an esp32

My thinking was that if the simple file length check worked I might expand it to a proper checksum. Anyway when I sent code.mpy to the esp32 the 'import code' line in boot.py promptly ran it before I'd even got around to writing the code to check & remove the 5 digit file length suffix.  

Other than discovering that .mpys should start with an M & that they compact .pys down to about 25% of their original size I'm otherwise ignorant. There appears to be no eof marker so curious if anybody knows why I can append a 5 digit number to a .mpy & the esp32 compiler is still happy to run it?

**Comment by bixb922** [comment, marked as answer]
See here https://docs.micropython.org/en/latest/reference/mpyfiles.html for an explanation of mpy files and their format.

The mpy file data is stored mainly using number of elements and their size. There seems not to be a eof marker. So extraneous appended data might be ignored for now. 

However I wouldn't rely on that. I suggest writing an additional json file with a dictionary of file names and their hash.
